docs/cpm.rst

```diff
@@ -0,0 +1,253 @@
+CPM Reference
+#############
+
+The `rapids_cpm` functions allow projects to find or build dependencies with built-in
+tracking of these dependencies for correct export support.
+
+rapids-cmake package defaults
+#############################
+
+rapids-cmake provides a collection of pre-defined packages to ensure that all consumers use
+the same version of dependencies. The exact versions that each pre-configured package uses
+can be found at the bottom of the page.
+
+.. literalinclude:: /packages/packages.rst
+
+rapids-cmake package override
+#############################
+
+rapids-cmake allows projects to override the default values for the pre-defined packages provided by rapids-cmake while also overriding any :cmake:command:`rapids_cpm_find`, `CPM <https://github.com/cpm-cmake/CPM.cmake>`_, and :cmake:module:`FetchContent() <cmake:module:FetchContent>` package calls.
+
+When an override for a project exists no :cmake:module:`find_package()<cmake:command:find_package>` search for that project will occur. This is done to make sure that the requested modified version is used.
+
+If a project is listed in multiple override files, the first file values will be used,
+and all later calls for that packaged will be ignored.  This "first to record, wins"
+approach is used to match FetchContent, and allows parent projects to override child
+projects.
+
+If the override is of an existing default project, it only needs to specify the
+fields it wants to override.
+
+For example if you wanted to change the version of `fmt` you would do the following:
+
+.. literalinclude:: /packages/example_git_tag_override.json
+  :language: json
+
+If you want to specify a completely new project, you need to specify at a minimum all
+the required fields:
+
+.. literalinclude:: /packages/example_new_project_override.json
+  :language: json
+
+
+reproducible rapids-cmake builds
+################################
+
+The rapids-cmake default `versions.json` uses branch names or git tag names
+for dependencies. This is done so that projects can 'live at head' of dependencies.
+This directly conflicts with the concept of reproducible release packages that each
+time they are built contain the exact same byteconde.
```

Comment:
There's a typo in bytecode you could fix instead if you prefer the original wording, but I think bytecode is a bit confusing to mention here since traditionally bytecode makes me think of interpreter IR (e.g. JVM or Python interpreter instructions) rather than machine code.
```suggestion
This directly conflicts with the concept of reproducible release packages that each
time they are built produce bitwise-identical artifacts.
```